src/code/z_eff_blure.c

```diff
@@ -66,40 +68,977 @@ void func_800A81F0(EffectBlure* this, Vec3f* p1, Vec3f* p2) {
+                vtx[j].v.cn[1] = sp60.g;
+                vtx[j].v.cn[2] = sp60.b;
+                vtx[j].v.cn[3] = sp60.a;
+            }
+        }
+
+        EffectBlure_DrawSimpleVertices(gfxCtx, this, vtx);
+    }
+}
+
+void EffectBlure_Draw(void* thisx, GraphicsContext* gfxCtx) {
+    EffectBlure* this = (EffectBlure*)thisx;
+    Vtx* vtx;
+    EffectBlureElement* elem;
+    s32 i;
+    s32 j;
+    s32 phi_t2;
+
+    OPEN_DISPS(gfxCtx);
+
+    gSPMatrix(POLY_XLU_DISP++, &D_801D1DE0, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+    if (this->numElements != 0) {
+        if (this->flags == 0) {
+            func_8012C560(gfxCtx);
+            gDPPipeSync(POLY_XLU_DISP++);
+
+            vtx = GRAPH_ALLOC(gfxCtx, sizeof(Vtx[32]));
+            if (vtx == NULL) {
+            } else {
+                j = 0;
+                for (i = 0; i < this->numElements; i++) {
+                    elem = &this->elements[i];
+
+                    if (elem->state == 1) {
+                        f32 ratio = (f32)elem->timer / (f32)this->elemDuration;
+
+                        switch (this->calcMode) {
+                            case 1:
+                                vtx[j].v.ob[0] = func_800B09D0(elem->p1.x, elem->p2.x, ratio);
+                                vtx[j].v.ob[1] = func_800B09D0(elem->p1.y, elem->p2.y, ratio);
+                                vtx[j].v.ob[2] = func_800B09D0(elem->p1.z, elem->p2.z, ratio);
+                                vtx[j + 1].v.ob[0] = elem->p2.x;
+                                vtx[j + 1].v.ob[1] = elem->p2.y;
+                                vtx[j + 1].v.ob[2] = elem->p2.z;
+                                break;
+                            case 2:
+                                vtx[j].v.ob[0] = elem->p1.x;
+                                vtx[j].v.ob[1] = elem->p1.y;
+                                vtx[j].v.ob[2] = elem->p1.z;
+                                vtx[j + 1].v.ob[0] = func_800B09D0(elem->p2.x, elem->p1.x, ratio);
+                                vtx[j + 1].v.ob[1] = func_800B09D0(elem->p2.y, elem->p1.y, ratio);
+                                vtx[j + 1].v.ob[2] = func_800B09D0(elem->p2.z, elem->p1.z, ratio);
+                                break;
+                            case 3:
+                                ratio = ratio * 0.5f;
+                                vtx[j].v.ob[0] = func_800B09D0(elem->p1.x, elem->p2.x, ratio);
+                                vtx[j].v.ob[1] = func_800B09D0(elem->p1.y, elem->p2.y, ratio);
+                                vtx[j].v.ob[2] = func_800B09D0(elem->p1.z, elem->p2.z, ratio);
+                                vtx[j + 1].v.ob[0] = func_800B09D0(elem->p2.x, elem->p1.x, ratio);
+                                vtx[j + 1].v.ob[1] = func_800B09D0(elem->p2.y, elem->p1.y, ratio);
+                                vtx[j + 1].v.ob[2] = func_800B09D0(elem->p2.z, elem->p1.z, ratio);
+                                ratio = ratio + ratio;
```

Comment:
```suggestion
                                ratio += ratio;
```
Is that clearer?

Reply:
I ended up changing it to *= 2.0f, which is the clearest option imo, and it's consistent with the *= 0.5f a few lines above.